knative-operator/pkg/webhook/knativekafka/webhook_validating_test.go

```diff
@@ -0,0 +1,120 @@
+package knativekafka_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	operatorv1alpha1 "github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis/operator/v1alpha1"
+	. "github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/knativekafka"
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/testutil"
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+)
+
+func init() {
+	configv1.AddToScheme(scheme.Scheme)
+	operatorv1alpha1.SchemeBuilder.AddToScheme(scheme.Scheme)
+}
+
+var ke1 = &operatorv1alpha1.KnativeKafka{
```

Comment:
These three KK's all look the same. Can we deduplicate? Maybe we can generalize the helpers from the other unit tests in a `testing` package in API and reuse here? 🤔

Reply:
yeah, sounds good. I will do this

Reply:
@markusthoemmes Do you mind if we leave these as is?

The function options approach doesn't scale well when things to mod are too many.

Reply:
Can we at least name them appropriately here then? I'm fine leaving as-is and potentially refactor later if necessary assuming that the names make it a bit more identifiable what each of them is supposed to represent.

I disagree on the scalability aspect as you can always pass a custom modifier function if you want to change multiple fields at once (upstream does that a lot). Not necessary to block the PR for that though, the naming would be enough imo.

Reply:
Ok, let's do that later. 

Renamed the CRs now. Actually added a "happy" test case too